getlatest.go

```diff
@@ -0,0 +1,135 @@
+package namesys
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	ggio "github.com/gogo/protobuf/io"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	pb "github.com/libp2p/go-libp2p-pubsub-router/pb"
+)
+
+type getLatestProtocol struct {
+	ctx  context.Context
+	host host.Host
+}
+
+func newGetLatestProtocol(ctx context.Context, host host.Host, getLocal func(key string) ([]byte, error)) *getLatestProtocol {
```

Comment:
We should probably make `getLocal` part of the type, given its usage pattern.
Also, better to typedef the function type.

Reply:
typedef 👍 (or an interface would also be fine)
part of the type 👎

If you'd prefer passing in an interface we can do that, but there's no reason to tie this to any particular implementation of "get stuff". We use a data store for this in the rest of this library, but data store has put + get, and we only need get.

Reply:
it's better to be part of the type i think...
no need for an interface, just a function typedef will do.

Reply:
done, although we'll need to revisit this if/when we expose this function publicly since an interface will likely be more appropriate.